src/viam/api/config/buf.gen.custom.remote.yaml.in

```diff
@@ -15,10 +15,10 @@
 version: v1
 plugins:
   - name: cpp
-    out: gen/
+    out: gen/viam/api
```

Comment:
A real change here though. When doing generation we *must* end up with a path that looks like `viam/api` so that include directives like `#include <viam/api/...` work. That does end up making a weird "doubled" path under the build directory, but that's ephemeral state, so I don't really care. Other buf.yaml files have the same change.

Reply:
I just realized that we can simplify this by removing all copying and this temporary `gen` directory.
all buf.gen.yaml files: 
```
   - name: cpp
-    out: gen/viam/api
+    out: .
   - name: cpp-grpc
-    out: gen/viam/api
+    out: .
```
api/CMakeLists.txt:
```
-  set(PROTO_GEN_DIR ${CMAKE_CURRENT_BINARY_DIR}/gen/viam/api)
+  set(PROTO_GEN_DIR ${CMAKE_CURRENT_BINARY_DIR})

   # Buf only goes back to 1.41.1 for generation against their "custom
   # remote" registry (see https://buf.build/grpc/plugins/cpp), and we
@@ -132,10 +132,6 @@ if (VIAMCPPSDK_USE_DYNAMIC_PROTOS)
     DEPENDS generate-dynamic-protos
   )

-  add_custom_command(
-    TARGET update-static-protos
-    COMMAND ${CMAKE_COMMAND} -E copy_directory ${PROTO_GEN_DIR} ${CMAKE_CURRENT_SOURCE_DIR}
-  )
```

This could allow us to remove the `PROTO_GEN_DIR` variable. I have tested this patch locally and things seem to be working.

Is there a reason that we kept this gen/... dir and then copy out of it? Sorry if I missed something!

Reply:
The first (getting rid of the `gen` and subpath) I think may be possible. At point point during the evolution of this PR I needed it for some reason, or thought I did. It is possible though that later changes have rendered that obsolete. I will consider it further.

The second part (removing the copy / PROTO_GEN_DIR) I don't think we can eliminate. When we are doing dynamic proto generation we are creating sources in the `build` tree, not the source tree. And we don't want to create them in the source tree since that overwrites the static protos which is annoying to deal with, since it means you always have edited files in your git state. That in fact is how it used to work, and moving to generating "to the side" was deliberate. So since we may or may not be building from static sources or dynamically generated ones, we need an abstraction and that is `PROTO_GEN_DIR`. Similarly, we need the copy command to be able to copy them back to the static sources when we want to.

Reply:
@zaporter-work - It took me a while to remember why I wanted make the `gen/viam/api` directory rather than generating to `.`. If you generate to `.` you are correct that everything works just fine at build time. But what you find is that when you do the directory copy to update the static protos, you end up copying intermediate cmake state out of the build directory into `src/viam/api` (e.g. files like `cmake_install.cmake` and `buf.gen.yaml`) when you do the directory copy to update the static protos. That's clearly undesirable. Pushing down into a `gen` diretory solves that because then you can just grab all the content of the `gen` directory, but when you do that you need `viam/api` too so that the includes still resolve right. I didn't love it when I typed it either.

I will add a TODO here that if we move to something where the exact file list of generated files is available for direct copy that we can remove the subdirectory thing.

Reply:
Ahh.. hence why I saw no problems because I had also removed the copy command. That makes sense. Good catch.